### ISSUE TITLE
[Accessibility] Enforce SemanticNode for embedded platform views not having other semantic information.

### ISSUE THREAD

**Issue by cyanglaz** [flutter/flutter#29396]
We should add code to enforce the semantic nodes for platform views (with platformViewId = null) to not have other semantic informations. 

To support a11y for embedded platform views, our logic ignores all the other semantic information when platformViewId is set. We have added documentation to explain this logic. 

To make this better, we should have code logic to enforce this. (e.g. using assertion)

It is a follow up refinement for https://github.com/flutter/flutter/issues/29302 and https://github.com/flutter/flutter/issues/29302

**Comment by cyanglaz** [comment]
Changed the implementation of SemanticObject so this is not required anymore.
